gui/src/app/components/TabWidget.tsx

```diff
@@ -0,0 +1,64 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { FunctionComponent, useState } from "react";
+import Tabs from "@mui/material/Tabs";
+import Tab from "@mui/material/Tab";
+import Box from "@mui/material/Box";
+
+type TabWidgetProps = {
+  labels: string[];
+  children: JSX.Element[];
+};
+
+const TabWidget: FunctionComponent<TabWidgetProps> = ({ labels, children }) => {
+  if (labels.length !== children.length) {
+    throw new Error("Number of labels and children must match");
+  }
+
+  const [index, setIndex] = useState(0);
+
+  const handleChange = (_: React.SyntheticEvent, newValue: number) => {
+    setIndex(newValue);
+  };
+
+  return (
+    <Box display="flex" height="100%" width="100%" flexDirection="column">
+      <Tabs value={index} onChange={handleChange}>
+        {labels.map((label, i) => (
+          <Tab key={i} label={label} />
+        ))}
+      </Tabs>
+      <Box flex="1" overflow="scroll">
+        {children.map((child, i) => (
+          <CustomTabPanel key={i} value={index} index={i}>
+            {child}
+          </CustomTabPanel>
+        ))}
+      </Box>
+    </Box>
+  );
+};
+
+interface TabPanelProps {
+  children?: React.ReactNode;
+  index: number;
+  value: number;
+}
+
+function CustomTabPanel(props: TabPanelProps) {
```

Comment:
explicit `function` style is different from what we use elsewhere (with arrow functions)